src/js/disability-benefits/526EZ/helpers.jsx

```diff
@@ -6,8 +6,41 @@ import { stateRequiredCountries } from '../../common/schemaform/definitions/addr
 import { transformForSubmit } from '../../common/schemaform/helpers';
 
 
+const siblings = ['treatments', 'privateRecordReleases', 'privateRecords', 'additionalDocuments'];
+
+export function flatten(data) {
+  const formData = Object.assign({}, data);
+  formData.disabilities.forEach((disability, idx) => {
+    siblings.forEach(sibling => {
+      if (disability[sibling]) {
+        formData[sibling] = [];
+        formData[sibling][idx] = disability[sibling];
+        delete disability[sibling]; // eslint-disable-line no-param-reassign
+      }
+    });
+  });
+  return formData;
+}
+
+export function nest(data) {
+  const formData = Object.assign({}, data);
+  siblings.forEach(sibling => {
+    if (formData[sibling]) {
+      formData[sibling].forEach((siblingData, idx, list) => {
+        formData.disabilities[idx][sibling] = siblingData; // eslint-disable-line no-param-reassign
+        delete formData[sibling][idx]; // eslint-disable-line no-param-reassign
+        if (idx === list.length - 1) {
+          delete formData[sibling]; // eslint-disable-line no-param-reassign
+        }
+      });
+    }
+  });
+  return formData;
+}
+
 export function transform(formConfig, form) {
-  const formData = transformForSubmit(formConfig, form);
+  let formData = transformForSubmit(formConfig, form);
+  formData = flatten(formData);
```

Comment:
What do you think of `const formData = flatten(transformForSubmit(formConfig, form));`?
Don't make no nevermind to me, but I thought I'd ask. :man_shrugging:

Reply:
Good idea, combining these function calls is easier to read (I'll double check that there is precedent for this). Can you imagine splitting it up facilitating debugging?

Reply:
`\(\w*\(\w*\)\)+` indicates this is a very common pattern, thanks for the heads up, have combined onto one line.